pinot-core/src/main/java/org/apache/pinot/core/util/GapfillUtils.java

```diff
@@ -71,12 +86,15 @@ public static boolean isFill(ExpressionContext expressionContext) {
       return false;
     }
 
-    return FILL.equals(canonicalizeFunctionName(expressionContext.getFunction().getFunctionName()));
+    return FILL.equalsIgnoreCase(canonicalizeFunctionName(expressionContext.getFunction().getFunctionName()));
```

Comment:
After #8341, all function name in `FunctionContext` is already canonical, no need to canonicalize again. Same for other places
```suggestion
    return expressionContext.getFunction().getFunctionName().equals(FILL);
```

Reply:
Fixed